src/pages/index.js

```diff
@@ -41,41 +56,157 @@ const addCardPopup = new PopupWithForm(
 );
 addCardPopup.setEventListeners();
 
+const editAvatarPopup = new PopupWithForm(
+  "#avatar-modal",
+  handleAvatarEditSubmit
+);
+editAvatarPopup.setEventListeners();
+
+const deleteConfirmation = new PopupWithConfirm("#delete-confirmation-modal");
+deleteConfirmation.setEventListeners();
+
 const cardPreviewPopup = new PopupWithImage("#image-modal");
 cardPreviewPopup.setEventListeners();
 
-const cardSection = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
-cardSection.renderItems(cardData);
+const user = new UserInfo({
+  name: ".profile__title",
+  job: ".profile__description",
+  avatar: ".profile__picture",
+});
 
-const user = new UserInfo(".profile__title", ".profile__description");
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "b7515436-5dfa-4f13-a09a-5183c1df5fb3",
+    "Content-Type": "application/json",
+  },
+});
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
 function handleProfileEditSubmit(profileData) {
-  const title = profileData.title;
-  const description = profileData.description;
-  user.setUserInfo(title, description);
-  editProfilePopup.close();
+  console.log("profileData:", profileData);
+  const name = profileData.title;
+  const job = profileData.description;
+  console.log("name:", name, "job:", job); //debugging
+  profileEditSubmitBtn.textContent = "Saving...";
+
+  api
+    .editProfile(profileData.title, profileData.description)
+    .then(() => {
+      user.setUserInfo({ name, job });
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditSubmitBtn.textContent = "Save";
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```